packages/flutter_tools/lib/src/runner/parser_extension.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/args.dart';
+
+extension ParserExtension on ArgParser {
```

Comment:
I'd remove this extension. You are writing much more boilerplate here than the code that you save by accepting an enum.

If this were something like:

```dart
void addEnumOption<T extends CoolEnum>(T value)
```

And compute the rest from `value`, I'd be onboard. But just to save a couple of calls to `map`, this seems excessive.